src/App.tsx

```diff
@@ -1,24 +1,162 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { addTodoToServer, getTodos, removeTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/todoList';
+import { TodoForm } from './components/todoForm';
+import { TodoFooter } from './components/todoFooter';
+import { Notification } from './components/notification';
+import { FilterBy } from './types/FilterBy';
 
-const USER_ID = 0;
+const USER_ID = 10221;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+  const [title, setTitle] = useState<string | undefined>('');
+  const [todoWasDeleted, setTodoWasDeleted] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disableForm, setDisableForm] = useState(false);
+  const [idProcessed, setIdProcessed] = useState<number>(0);
+  const [disableList, setDisableList] = useState(false);
+
+  function getVisibleTodos(filterType: FilterBy) {
+    switch (filterType) {
+      case FilterBy.ACTIVE:
+        setVisibleTodos(todos.filter((todo: Todo) => !todo.completed));
+        break;
+
+      case FilterBy.COMPLETED:
+        setVisibleTodos(todos.filter((todo: Todo) => todo.completed));
+        break;
+
+      default:
+        setVisibleTodos(todos);
+    }
+  }
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((fetchedTodos: Todo[]) => {
+        setTodos(fetchedTodos);
+        setTodoWasDeleted(false);
+      })
+      .catch((fetchedError) => {
+        setError(
+          fetchedError?.message
+            ? fetchedError.message
+            : 'Something went wrong',
+        );
+
+        setTodoWasDeleted(false);
+      });
+  }, [todoWasDeleted === true]);
```

Comment:
The dependency array in this useEffect should contain only dependencies, not conditions. Consider changing it to [todoWasDeleted] rather than [todoWasDeleted === true].